test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -733,3 +734,206 @@ spec:
+		t.Fatal(err)
+	}
+	for i := range noxuDefinition.Spec.Versions {
+		noxuDefinition.Spec.Versions[i].Schema = &c
+	}
+
+	noxuDefinition, err = fixtures.CreateNewV1CustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kind := noxuDefinition.Spec.Names.Kind
+	// Create a CR in the current stored version, i.e. v1beta2.
+	apiVersion := noxuDefinition.Spec.Group + "/" + noxuDefinition.Spec.Versions[1].Name
+	name := "mytest"
+
+	rest := apiExtensionClient.Discovery().RESTClient()
+	yamlBody := []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+spec:
+  cronSpec: "* * * * */5"
+  replicas: 1
+  ports:
+  - name: x
+    containerPort: 80
+    protocol: TCP`, apiVersion, kind, name))
+	result, err := rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Versions[1].Name, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw(context.TODO())
+	if err != nil {
+		t.Fatalf("failed to create custom resource with apply: %v:\n%v", err, string(result))
+	}
+
+	crdClient := apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions()
+	// Migrate away from current storage version of v1beta2 to v1beta1.
+	crd, err := crdClient.Get(context.TODO(), noxuDefinition.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get CRD: %v", err)
+	}
+	crd.Spec.Versions[1].Storage = false
+	crd.Spec.Versions[0].Storage = true
+
+	newCRD, err := crdClient.Update(context.TODO(), crd, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("failed to update CRD: %v", err)
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    newCRD.Spec.Group,
+		Version:  newCRD.Spec.Versions[1].Name,
+		Resource: newCRD.Spec.Names.Plural,
+	}
+	// Migrate existing CR to new storage version by applying an
+	// empty merge patch.
+	_, err = dynamicClient.Resource(gvr).
+		Patch(context.TODO(), name, types.MergePatchType, []byte("{}"), metav1.PatchOptions{})
+	if err != nil {
+		t.Fatalf("failed to patch CR: %v", err)
+	}
+
+	// Remove old storage version from status
+	patchedCRD, err := crdClient.Patch(context.TODO(),
+		newCRD.Name,
+		types.MergePatchType,
+		[]byte(`{"status":{"storedVersions":["`+newCRD.Spec.Versions[0].Name+`"]}}`),
+		metav1.PatchOptions{}, "status",
+	)
+	if err != nil {
+		t.Fatalf("failed to patch CRD: %v", err)
+	}
+	crd, err = crdClient.Get(context.TODO(), patchedCRD.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get CRD: %v", err)
+	}
+
+	// Remove the old version from spec.versions
+	crd.Spec.Versions = append(crd.Spec.Versions[:1], crd.Spec.Versions[2:]...)
+	_, err = crdClient.Update(context.TODO(), crd, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("failed to update CRD: %v", err)
+	}
+
+	apiVersion = noxuDefinition.Spec.Group + "/" + noxuDefinition.Spec.Versions[0].Name
+	yamlBody = []byte(fmt.Sprintf(`
+apiVersion: %s
```

Comment:
Make sure you send new fields in that apply so that you can see from the list of managed fields exactly what happened. (i.e. the old entry was removed even though it still should have owned some fields), the second entry that was inserted also still exists, and that last one also exist. You should have two entries at the end.